.github/workflows/ci.yaml

```diff
@@ -39,7 +39,15 @@ jobs:
       - name: Check Doc Modified
         id: check_doc_modified
         run: |
-          python -m pkgmt.fail_if_modified -b origin/master -e doc 
+          if [ "$GITHUB_EVENT_NAME" == "pull_request" ]
+          then
+            echo "Pull request, running check_doc"
+            python -m pkgmt.fail_if_modified -b origin/master -e doc 
+          else
+            echo "This is not a pull request event"
+            echo "Running all tests"
+            exit 1
```

Comment:
I guess `exit 1` will make the action failure?
Should we do `exit 0`?

[Ref](https://docs.github.com/en/actions/creating-actions/setting-exit-codes-for-actions#about-exit-codes)

Reply:
To run all tests, we want `check_doc` to fail. The test main purpose to determine whether anything other than `doc` has been modified. The test will fail if anything other than `doc` was modified and will run all tests.

Reply:
OK just saw `continue-on-error: true`
So I guess that still marks the failing step as "succeeded"?

Reply:
`continue-on-error: true` make sure that the jobs below will run irrespective of the failure or success of a step. 

I am using the statement the outcome of the step to determine whether the next test will run or not.
`if: needs.check_doc.outputs.check_doc_modified == 'failure'`

Reply:
I see, logically it looks good to me.
Need to merge to master to verify